### DISCUSSION TITLE
PULL section does not function

### DISCUSSION THREAD

**Discussion by EmiliaGS** [BHoM/BHoM#1420] [Q&A]
Hi! 
I have written down the whole script and went through the video tutorial of Kayleigh Houde a few times. 
https://www.autodesk.com/autodesk-university/class/Using-Revit-and-Dynamo-Assess-Embodied-Carbon-2020#video

However the PULL function simply shows a empty list and I do not understand why. It says the input from the Adapter is missing even thought it is clearly connected. 
I am using Revit 2020 with Dynamo. 

Thank you in advance for your guidance! 

![Pull_BHom_20220812](https://user-images.githubusercontent.com/110984780/184340832-96b93582-37ec-49dc-9572-7f30839d346d.png)
![image](https://user-images.githubusercontent.com/110984780/184352009-00fc388c-a732-4068-8057-7e5467a722ba.png)

**Comment by FraserGreenroyd** [comment]
@pawelbaran @vietle-bh @kayleighhoude fyi

**Reply by EmiliaGS** [reply]
Thank you for your fast responce!

**Comment by pawelbaran** [comment]
Hi @EmiliaGS - thank you for the question! Apparently, it looks like the `Pull` component is missing the `adapter` input. Could you please start from setting `RevitAdapter.Active` to `true` and checking whether the output is `null`? Dynamo UI is a bit clunky so that might be the problem. In general, it is highly recommended to use BHoM with Grasshopper instead of Dynamo 😉

**Reply by EmiliaGS** [reply]
Hi @pawelbaran 
The RevitAdapter.Active is set to True with the CodeBlock as shown in the screenshot, and the output is still null...
Highly appreciate your answer! I have switched to Grasshopper :)

**Reply by pawelbaran** [reply]
> I have switched to Grasshopper :)

This is best you can do to be honest @EmiliaGS - we barely support Dynamo in the recent years (as you may have noticed, BHoM does not work in Dynamo under Revit 2021 and above), while Grasshopper is under continuous maintenance and development. Please do not hesitate to reach out to us in case of any further questions or feedback, thanks!

**Reply by EmiliaGS** [reply]
Hi @pawelbaran, 
I have switched to Grasshopper and get the following error that I do not understand. Could you please clarify why object reference are not set to an instance of an object? Thank you in advance for your support and guidance! 
- Occured in  BH.Revit.Engine.Core.Query.ElementIds(FilterRequest request, Document document, Discipline discipline, RevitSettings settings, IEnumerable`1 ids)
     called from BH.Revit.Engine.Core.Query.IElementIds(IRequest request, Document document, Discipline discipline, RevitSettings settings, IEnumerable`1 ids)
- Are you sure you have the correct type of inputs? Check their description for more details.

Here is the full text
![GH_BHom_Error PULL](https://user-images.githubusercontent.com/110984780/185336335-a32ecb55-f31a-4857-ad6d-bde17a295152.png)

**Reply by pawelbaran** [reply]
Hi @EmiliaGS, thanks for reaching out again. In general, null reference exception means that a tried to process a `null` value, which usually occurs when the user passes it as an input. I just tried to reproduce the error on my machine, without luck - it all works fine, or if I try really hard, it throws a different (better formatted) exception at the user.

Which version of BHoM are you using? Also is there a chance you could share the .gh script you are working with? Happy to have another look on setup as close to yours as possible 👍

**Reply by EmiliaGS** [reply]
[BHom_Test_Tutorial_EmiliaStefanova.zip](https://github.com/BHoM/BHoM/files/9379918/BHom_Test_Tutorial_EmiliaStefanova.zip)
Hi @pawelbaran, here is the script, would really appreciate if you could have a look at it as I think I have followed the tutorial exactly as it is. Now even tested to set another type of FilterRequest that you will see. Again, really appreciate your guidance!

**Reply by pawelbaran** [reply]
This one of those we like the most - it works like a charm on my machine 😉 Could you please tell me which version of BHoM and Revit are you using? Also now I realised it is run from Rhino Inside Revit, am I correct? Finally, could you try running the script on the sample arch Revit model, so that we could check it occurs repetitively on all models on your end? Hopefully we can get to the bottom of the issue based on that info. Again, apologies for the requested effort, but this is a really interesting case that we have never experienced in the past - I am in the investigation mode!

![image](https://user-images.githubusercontent.com/26874773/185597282-b0b0d490-4461-42c7-8701-1c78a0c20667.png)

**Reply by EmiliaGS** [reply]
Ok, finally I tested the script against the racbasicsampleproject.rvt and it works! Thank you for the tip! 
And it was no effort at all, as I have been trying to figure it out as well. Just did not though it could be because of the model itself. Investigation mode is my favorite as well :D 
Othwerwise: 
BHom: 5.2.β.0 
Revit: 2020.2 & 2022 ( it works here too)
Rhino.Inside.Revit: v1.0 You guessed it right 😊 
Have a nice weekend Pawel!
